services/datalad/tests/test_snapshots.py

```diff
@@ -39,6 +40,29 @@ def test_create_snapshot(client, new_dataset):
     assert response.status == falcon.HTTP_OK
 
 
+def test_create_snapshot_no_config(datalad_store, client, new_dataset):
+    """Validate adding a datalad config if one is missing during snapshot creation"""
+    ds_id = os.path.basename(new_dataset.path)
+    snapshot_id = '1'
+    # Delete the default config first
+    response = client.simulate_delete('/datasets/{}/files'.format(
+        ds_id), body='{ "filenames": [".datalad/config"] }')
+    assert response.status == falcon.HTTP_OK
+    assert json.loads(response.content)['deleted'] == [
+        '.datalad/config']
```

Comment:
Since our later check is that Datalad could find an ID, I think it would make sense to verify that it can't pre-snapshot.

```suggestion
        '.datalad/config']
    ds = Dataset(os.path.join(datalad_store.annex_path, ds_id))
    assert ds.id is None
```

Reply:
Added a variant of this but have to close the dataset or datalad caches the id despite removing the config.